spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -842,7 +842,7 @@ def resample_labels(fname_labels, fname_dest, fname_output):
                               int(float(v))])
                   for x, y, z, v in og_labels]
 
-    sct_labels.create_labels_empty(Image(fname_dest), new_labels).save(path=fname_output)
+    sct_labels.create_labels_empty(Image(fname_dest).change_type('uint8'), new_labels).save(path=fname_output)
```

Comment:
Do we _always_ want labels to be integers? (i.e. there is no valid case where labels should be float)

If this is true, then perhaps this change should be applied within the `create_labels_empty` or `_add_labels`  functions instead.

Reply:
i've asked myself the same question. So far, all our labels are indeed integer only. I didn't want to be "potentially" disruptive in this PR because i do not have the time right now to investigate potential collateral damage-- but your suggestion is excellent-- since we need this pr to be merged asap i suggest you open an issue to take the time to look into making the change in the API instead-- many thanks! ps --> i will not have the time today to look into this issue-- so can i pls ask you to approve if it looks ok-- then merge-- thanks!

Reply:
> so can i pls ask you to approve if it looks ok-- then merge--

Sounds good! I can confirm that your change does fix the issue, so I'll approve.

I will also unlink this PR from the issue so it doesn't auto-close. The root cause is something bigger (related to `Image.save()` functionality), but that can be addressed in a separate PR as you mention.